### PR TITLE
[Test] Fills Item

### PR DESCRIPTION
**Pull request by Alecaddd**
This is the initial setup for the Vala Unit Test folder.
The scope of this PR is to provide a basic environment to run tests, and land the first set of tests covering some basic Canvas interactions.

**What's implementing**
- [x] Run tests by typing `meson test` from `/build` folder.
- [x] Setup test environment on first run.
- [x] Initialize Canvas and generate 1000 items.
- [x] Quickly select and deselect the all the available items.
- [x] Listen to the FillsListItem model and assert no segfault occurs.

### CODE REVIEW

**Comment by Alecaddd** [comment]
Hey guys, whenever you have time, could you take a look at the current setup for the tests?
I'm able to run the simple test I created successfully locally, but it seems that GItHub CI is not able to run GUI tests as I'm getting these errors.
`Unable to init server: Could not connect: Connection refused`
`(simple_unit_test:1053): Gtk-WARNING **: 01:39:44.246: cannot open display:`

Let me know if you have any clue on how to fix this, if it's even possible to fix.
@bilelmoussaoui @albfan @giacomoalbe

**Comment by Alecaddd** [comment]
All right, this is ready for a full review.
I temporarily disabled the `meson test` run for the GitHub CI as it will always fail because of the lack of a display manager.
Once this lands, I will implement Xvfb to run these tests in a Headless way.

For now, you can pull this branch and run your the test locally by running `meson test` from your `/build` folder.

Apologies for all the file changes, as I've fixed the linting warnings throughout all the files.

**Comment by Alecaddd** [comment]
I fixed it!
Now all the tests can run in the CI at every push or pull request.

File: tests/tests.vala [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+    public GLib.TestCase get_g_test_case () {
+        return new GLib.TestCase (this.name,
+                                  this.setup,
+                                  this.run,
+                                  this.teardown);
+    }
+}
+
+public abstract class Akira.TestSuite : GLib.Object {
```
Comment (albfan) [inline tests/tests.vala]: This do not need to be abstract. It complains on new vala versions

Comment (Alecaddd) [reply tests/tests.vala]: Updated

**Pull request merged**